### ISSUE TITLE
a11y: improve the button to change location

### ISSUE THREAD

**Issue by afercia** [coreymckrill/nearby-wordpress-events#34]
The button to change nearby location could be improved a bit for a11y:

<img width="657" alt="screen shot 2017-04-17 at 17 16 12 2" src="https://cloud.githubusercontent.com/assets/1682452/25093919/e9138352-2394-11e7-8e6a-1ee2aef4582b.png">

Currently, the only button text is the name of the location. That doesn't help so much when the button text gets read out of context or announced by screen readers. The button performs an action (opens the form to edit the location) and that action should be reflected in the button text.

**Comment by coreymckrill** [comment]
@afercia Thanks. I'm trying to think of how to keep the location name clickable while providing more context for screen readers. Do you think something like this would be an improvement?

```
<button id="nearbywp-toggle" class="button-link nearbywp-toggle">
	<strong>{{{ data.location.description }}}</strong>
	<span class="screen-reader-text"><?php esc_html_e( '(Change the location)', 'nearby-wp-events' ); ?></span>
	<span class="dashicons dashicons-edit" aria-hidden="true"></span>
</button>
```

**Comment by iandunn** [comment]
Note that the `{{{` was changed to `{{` in 26d1b5ace200b0edd35344a440d2a09f496c8690, so we need to make sure that's not overwritten in this change. I'm not sure if `git push` would abort or not if a change was pushed before pulling in 26d1b5ace200b0edd35344a440d2a09f496c8690. Subversion wouldn't, so I figured I'd be careful and make a note here.

**Comment by afercia** [comment]
@coreymckrill yep if you want to keep the location name clickable, that could work. Personally, I wouldn't do that though. I'd probably go with something similar to what is already used, for example, in the Publish box. A visible `Edit` text with some expanded text for assistive technologies. The difference here is the control is a button (that would mean also: no icon).
```
Attend an upcoming event near <strong>{{{ data.location.description }}}</strong>.
<button id="nearbywp-toggle" class="button-link nearbywp-toggle" aria-label="<?php esc_attr_e( 'Edit location', 'nearby-wp-events' ); ?>">
	<?php esc_html_e( 'Edit', 'nearby-wp-events' ); ?>
	<span class="dashicons dashicons-edit" aria-hidden="true"></span>
</button>

```

Also, I've just realized the button expands the small panel below so it needs an `aria-expanded` attribute, will open a separate issue 🙂

**Comment by iandunn** [comment]
@mapk, how do you feel about replacing the icon with an explicit label?

![screen shot of widget with label instead of icon](https://cloud.githubusercontent.com/assets/484068/25206648/26929224-251e-11e7-8f05-e8bb139e8296.png)

I wonder if that might be more intuitive for all users, in addition to the a11y benefits?

**Comment by mapk** [comment]
I'm preferable to how we had it if we can get it to work with a11y. I know we use edit buttons in other places too like here:

![screen shot 2017-04-19 at 6 41 36 pm](https://cloud.githubusercontent.com/assets/617986/25209434/ea6f2520-252f-11e7-90eb-b39bb6d886fe.png)

Maybe it can be a small edit icon next to the word "Edit"  ... and not in parenthesis?

**Comment by coreymckrill** [comment]
If we move the name of the location out of the button, I think things get easier. Then we can still have the edit icon, and have the "Edit Location" text be hidden except for screen readers. Sort of a combination of my suggestion and @afercia 's, like this:

```
Attend an upcoming event near <strong>{{{ data.location.description }}}</strong>.
<button id="nearbywp-toggle" class="button-link nearbywp-toggle" aria-label="<?php esc_attr_e( 'Edit location', 'nearby-wp-events' ); ?>">
	<span class="screen-reader-text"><?php esc_html_e( 'Edit location', 'nearby-wp-events' ); ?></span>
	<span class="dashicons dashicons-edit" aria-hidden="true"></span>
</button>
```

**Comment by afercia** [comment]
Yep, in this case `aria-label` makes the `screen-reader-text` redundant, as it overrides the content of the button 🙂

**Comment by coreymckrill** [comment]
Ah, I wondered about that. I'll work up a commit with the above, minus the `screen-reader-text` span.
